celery_haystack/utils.py

```diff
@@ -10,20 +10,24 @@
 from .conf import settings
 
 
-def get_update_task(task_path=None):
-    import_path = task_path or settings.CELERY_HAYSTACK_DEFAULT_TASK
-    module, attr = import_path.rsplit('.', 1)
+def get_class(import_path):
+    module_name, attr = import_path.rsplit('.', 1)
     try:
-        mod = import_module(module)
+        mod = import_module(module_name)
     except ImportError as e:
         raise ImproperlyConfigured('Error importing module %s: "%s"' %
-                                   (module, e))
+                                   (module_name, e))
     try:
-        Task = getattr(mod, attr)
+        update_task = getattr(mod, attr)
     except AttributeError:
         raise ImproperlyConfigured('Module "%s" does not define a "%s" '
-                                   'class.' % (module, attr))
-    return Task()
+                                   'class.' % (module_name, attr))
+    return update_task
+
+
+def get_update_task(task_path=None):
+    import_path = task_path or settings.CELERY_HAYSTACK_DEFAULT_TASK
+    return get_class(import_path)
```

Comment:
Are you sure this is working?  The former `get_update_task` returns an instantiated task, but this returns the class.

Reply:
It seems like he left that part alone in the latest commit of his fork:
https://github.com/dwintergruen/celery-haystack/blame/develop/celery_haystack/utils.py

I am going to use his library to see if it works, but it seems like it should
@ddemid Do you plan to make a PR with your latest commit?

Reply:
@grjones Yes, I'm sure. This version is supposed to work with ```@shared_task``` which is a function as it's recommended in the [Celery documentation](http://docs.celeryproject.org/en/latest/django/first-steps-with-django.html#using-the-shared-task-decorator).

I'm not a big fan of huge class-based tasks which are harder to test. 
Custom functionality should live in the handler in my opinion.